src/data/list/basic.lean

```diff
@@ -1961,6 +1961,10 @@ if_pos rfl
 @[simp] theorem count_cons_of_ne {a b : α} (h : a ≠ b) (l : list α) : count a (b::l) = count a l :=
 if_neg h
 
+theorem count_tail : Π (l : list α) (a : α) (h : 0 < l.length),
+  l.tail.count a = l.count a - ite (a = list.nth_le l 0 h) 1 0
+| (_ :: _) a h := by { rw [count_cons], split_ifs; simp }
+
```

Comment:
Not sure, but maybe this should be `l.tail.count a + ite (a = list.nth_le l 0 h) 1 0 = l.count a` just to avoid natural number subtraction.

Reply:
I thought about this, but it's really intended as a simplification lemma, so I wanted the LHS to be the thing being calculated. If I move the `ite` to the LHS, at least in my use case, and probably others, the user is just going to have to rearrange the lemma by hand before they can use it.

Reply:
Relatedly, I wouldn't mind some nice notation for `ite P 1 0`.

Reply:
The nice notation is `if P then 1 else 0`. What would be nicer?

Reply:
I would be okay with a definition like `boole P` or similar to mean `ite P 1 0`. But it doesn't save a lot of space, and I'm not sure how many interesting theorems there are to prove about it.